src/sources/internal_metrics.rs

```diff
@@ -44,12 +33,16 @@ impl SourceConfig for InternalMetricsConfig {
         shutdown: ShutdownSignal,
         out: Pipeline,
     ) -> crate::Result<super::Source> {
-        Ok(Box::pin(run(
-            get_controller()?,
-            self.scrape_interval_secs,
-            out,
-            shutdown,
-        )))
+        let interval = time::Duration::from_secs_f64(self.scrape_interval_secs);
+        if interval < time::Duration::from_millis(500) {
+            return Err(format!(
+                "interval set too low ({} secs), use interval >= 0.5 secs",
```

Comment:
Assuming we want to keep this a float, I'm wondering if this should be a warning rather than an error as there is nothing technically stopping them from using a lower value, it just may increase CPU utilization.

Reply:
Well, this is reasonable, I'll change to warning.

Reply:
Done